common/app/common/cms/CmsService.java

```diff
@@ -0,0 +1,19 @@
+package common.cms;
+
+import play.libs.F;
+
+import java.util.Locale;
+
+/**
+ * Service that provides page content, usually coming from a Content Management System (CMS).
+ */
+public interface CmsService {
+
+    /**
+     * Gets the page content corresponding to the given key for a certain language.
+     * @param locale for the localized text
+     * @param pageKey identifying the page
+     * @return the promise of the page content identified by the key, in the given language
+     */
+    F.Promise<CmsPage> get(final Locale locale, final String pageKey);
```

Comment:
Yes, probably we'll need more optional arguments, but I would need to implement the Contentful implementation first to see a better overview. But I think as a first draft for the POC is good enough, the additional changes won't be that costly.

About the `Option`, for the page I think it's more adequate to throw an exception if the page is not found. There's not many things you can do without the content. A different approach is taken for a single message that cannot be found, there an `Optional` is returned.